src/ngraph_sgcompiler.cc

```diff
@@ -61,6 +79,15 @@ void CompileForward(std::shared_ptr<Graph> sub_graph,
     dump_graph(f, __func__, "fprop_compiled");
   }
   sub_graph->ngraph_forward[mode] = f;
+
+  sub_graph->is_bool_[mode][(int)(NodeReferences::kForwardInput)] =
+      ngraph_bridge::is_bool(f->get_parameters());
+  sub_graph->is_scalar_[mode][(int)(NodeReferences::kForwardInput)] =
+      ngraph_bridge::is_scalar(f->get_parameters());
+  sub_graph->is_bool_[mode][(int)(NodeReferences::kForwardOutput)] =
+      ngraph_bridge::is_bool(f->get_results());
+  sub_graph->is_scalar_[mode][(int)(NodeReferences::kForwardOutput)] =
+      ngraph_bridge::is_scalar(f->get_results());
```

Comment:
NBD, but IMHO it would be nice to have a little more [psychological distance](https://books.google.com/books?id=pDsFCAAAQBAJ&pg=PA556&lpg=PA556&dq=%22variable+names%22+%22psychological+distance%22&source=bl&ots=RmJoIDzvUT&sig=bMXQh4o_C4ZSTI67KT3ywzHD8wY&hl=en&sa=X&ved=2ahUKEwjks9SZ8JzeAhWYITQIHQVkCccQ6AEwAHoECAkQAQ#v=onepage&q=%22variable%20names%22%20%22psychological%20distance%22&f=false) between the names `is_bool` / `is_scalar` and `is_bool_` / `is_scalar_`.

Reply:
:) This bit of code is ugly. Will think about a cleaner solution